src/server.rs

```diff
@@ -198,6 +201,10 @@ impl Endpoint {
         self.clients.keys().cloned().collect()
     }
 
+    /// Attempt to deserialise a message into type `T`.
+    ///
+    /// Will return [`Err`] if the bytes accumulated from the client aren't deserializable to T.
+    /// This could be from `T` not being what the client sent, or because the server has not received enough bytes to fully deserialize the message.
```

Comment:
`receive_message_from` will always return a full message, so you can remove "or because the server has not received enough bytes to fully deserialize the message."

```suggestion
    /// Will also return [`Err`] if this client is disconnected.
```

Reply:
What is the error if there was insufficient bytes to successfully deserialise. I assume just generic deserialise error? So deserialiser doesn't know/care if there was enough bytes.

Will remove that section 👍

Reply:
This error cannot happen at this level since the client/server async back-end only transmits full payloads/messages to the sync client/server.